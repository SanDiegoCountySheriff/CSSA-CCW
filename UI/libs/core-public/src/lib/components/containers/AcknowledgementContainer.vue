<template>
  <v-container>
    <v-card
      v-if="step.index === 0"
      class="acknowledgement-card"
    >
      <v-card-title> </v-card-title>
      <v-card-text>
        <AcknowledgementInitial
          :handle-accept="handleAccept"
          :handle-decline="handleDecline"
        />
      </v-card-text>
    </v-card>

    <v-card
      v-if="step.index === 1"
      class="acknowledgement-card"
    >
      <v-card-title> {{ $t('acknowledgement-title-part2') }} </v-card-title>
      <v-card-text>
        <AcknowledgementPart
          :handle-accept="handleAccept"
          :handle-decline="handleDecline"
          :text-body="'acknowledgement-part1'"
        />
      </v-card-text>
    </v-card>

    <v-card
      v-if="step.index === 2"
      class="acknowledgement-card"
    >
      <v-card-title> {{ $t('acknowledgement-title-part3') }} </v-card-title>
      <v-card-text>
        <AcknowledgementPart
          :handle-accept="handleAccept"
          :handle-decline="handleDecline"
          :text-body="'acknowledgement-part2'"
        />
      </v-card-text>
    </v-card>

    <v-card
      v-if="step.index === 3"
      class="acknowledgement-card"
    >
      <v-card-title> {{ $t('acknowledgement-title-part4') }} </v-card-title>
      <v-card-text>
        <AcknowledgementPart
          :handle-accept="handleAccept"
          :handle-decline="handleDecline"
          :text-body="'acknowledgement-part3'"
        />
      </v-card-text>
    </v-card>

    <v-card
      v-if="step.index === 4"
      class="acknowledgement-card"
    >
      <v-card-title> {{ $t('acknowledgement-title-part5') }} </v-card-title>
      <v-card-text>
        <AcknowledgementPart
          :handle-accept="handleAccept"
          :handle-decline="handleDecline"
          :text-body="'acknowledgement-part4'"
        />
      </v-card-text>
    </v-card>

    <v-card
      v-if="step.index === 5"
      class="acknowledgement-card"
    >
      <v-card-title> {{ $t('acknowledgement-title-part6') }} </v-card-title>
      <v-card-text>
        <AcknowledgementPart
          :handle-accept="handleAccept"
          :handle-decline="handleDecline"
          :text-body="'acknowledgement-part5'"
          :link="Routes.PENAL_CODE_PATH"
        />
      </v-card-text>
    </v-card>

    <v-card
      v-if="step.index === 6"
      class="acknowledgement-card"
    >
      <v-card-title> {{ $t('acknowledgement-title-part7') }} </v-card-title>
      <v-card-text>
        <AcknowledgementPart
          :handle-accept="handleAccept"
          :handle-decline="handleDecline"
          :text-body="'acknowledgement-part6'"
        />
      </v-card-text>
    </v-card>

    <v-card
      v-if="step.index === 7"
      class="acknowledgement-card"
    >
      <v-card-title> {{ $t('acknowledgement-title-part8') }} </v-card-title>
      <v-card-text>
        <AcknowledgementPart
          :handle-accept="handleFinalAccept"
          :handle-decline="handleDecline"
          :text-body="'acknowledgement-part7'"
        />
      </v-card-text>
    </v-card>
  </v-container>
</template>

<script setup lang="ts">
import AcknowledgementInitial from '@shared-ui/components/acknowledgement-section/AcknowledgementInitial.vue';
import AcknowledgementPart from '@shared-ui/components/acknowledgement-section/AcknowledgementPart.vue';
import Routes from '@core-public/router/routes';
import { reactive } from 'vue';
import { useCompleteApplicationStore } from '@shared-ui/stores/completeApplication';
import { useRouter } from 'vue-router/composables';

interface AcknowledgementProps {
  nextRoute: string;
}

const router = useRouter();
const applicationStore = useCompleteApplicationStore();
const step = reactive({
  index: 0,
});
const props = defineProps<AcknowledgementProps>();

function handleAccept() {
  step.index += 1;
}

function handleDecline() {
  router.push('/');
}

function handleFinalAccept() {
  applicationStore.completeApplication.application.currentStep = 1;
  router.push({
    path: props.nextRoute,
    query: {
      orderId: applicationStore.completeApplication.application.orderId,
      isComplete: applicationStore.completeApplication.application.isComplete,
    },
  });
}
</script>

<style lang="scss" scoped>
.acknowledgement {
  &-card {
    border-radius: 0.5rem;
    border: 0.75rem solid rgba(29, 41, 57, 1);
  }
}
</style>
