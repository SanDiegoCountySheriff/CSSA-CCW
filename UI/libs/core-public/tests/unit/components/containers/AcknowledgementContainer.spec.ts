/* eslint-disable @typescript-eslint/ban-ts-comment */
import AcknowledgementContainer from '@core-public/components/containers/AcknowledgementContainer.vue';
import Vuetify from 'vuetify';
import { createTestingPinia } from '@pinia/testing';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
const pinia = createTestingPinia();
const tMock = {
  $t: text => text,
};

describe('AcknowledgementContainer', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(AcknowledgementContainer, {
      localVue,
      pinia,
      vuetify,
      mocks: tMock,
      propsData: {
        nextRoute: 'test',
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
