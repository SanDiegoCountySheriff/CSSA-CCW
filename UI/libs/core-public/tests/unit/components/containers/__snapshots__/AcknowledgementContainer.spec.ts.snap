// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcknowledgementContainer Should match the snapshot 1`] = `
<div class="container">
  <div class="acknowledgement-card v-card v-sheet theme--light">
    <div class="v-card__title">Acknowledgements</div>
    <div class="v-card__text">
      <div class="container">
        <div class="container acknowledgement-container">
          <div class="row">
            <div class="text-left acknowledgement-text col">
              acknowledgement-initial-part1
            </div>
          </div>
          <div class="row">
            <div class="text-left acknowledgement-text col">
              acknowledgement-initial-part2
            </div>
          </div>
          <div class="row">
            <div class="text-left acknowledgement-text col">
              acknowledgement-initial-part3
            </div>
          </div>
          <div class="row">
            <div class="text-left acknowledgement-text col">
              acknowledgement-initial-part4
            </div>
          </div>
          <div class="row">
            <div class="text-left acknowledgement-text col">
              acknowledgement-initial-part5
            </div>
          </div>
        </div>
        <div class="button-container">
          <div class="acknowledgement-button-container"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default success mr-3"><span class="v-btn__content">
    Accept
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error mr-3"><span class="v-btn__content">
    Decline
  </span></button></div>
        </div>
      </div>
    </div>
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;
